compiler/jsgen.nim

```diff
@@ -1655,15 +1655,15 @@ proc genVarInit(p: PProc, v: PSym, n: PNode) =
     varCode: string
     varName = mangleName(p.module, v)
     useReloadingGuard = sfGlobal in v.flags and p.config.hcrOn
-
+  const jsVarDecl = when defined(nimJsVar): "var" else: "let"
```

Comment:
this makes no sense. Why do you need to recompile nim (aka different dialects) just to change nim's behavior?
instead either use `isDefined(conf, "nimJsVar")` or, better, use the suggestion i gave above with `--legacy:jsUseVar`

Reply:
I like it compile time, because either it always works or never works,
and I dont see people switching back to `var` if compiler uses `let` already.
But ok.